mm2src/coins/eth/v2_activation.rs

```diff
@@ -215,11 +251,85 @@ impl EthCoin {
+    /// including required confirmations, web3 instances, and NFT information fetched from a provided URL.
+    ///
+    /// # Arguments
+    ///
+    /// * `ctx`: Shared application context.
+    /// * `chain`: Blockchain to which the NFTs belong.
+    /// * `conf`: Configuration parameters for the coin.
+    /// * `activation_request`: Activation request containing specific requirements.
+    /// * `url`: URL to fetch NFT data from.
+    ///
+    /// # Returns
+    ///
+    /// * `MmResult<EthCoin, EthActivationV2Error>`: Resulting `EthCoin` instance for global NFTs, or an error.
+    pub async fn global_nft_from_platform_coin(
+        &self,
+        ctx: &MmArc,
+        chain: &Chain,
+        conf: &Json,
+        activation_request: &EthActivationV2Request,
+        url: &Url,
+    ) -> MmResult<EthCoin, EthActivationV2Error> {
+        let ticker = chain.to_nft_ticker().to_string();
+
+        // Create an abortable system linked to the `MmCtx` so if the app is stopped on `MmArc::stop`,
+        // all spawned futures related to global non fungible token will be aborted as well.
+        let abortable_system = ctx.abortable_system.create_subsystem()?;
+
+        // param from request should override the config
+        let required_confirmations = activation_request
+            .required_confirmations
+            .unwrap_or_else(|| {
+                conf["required_confirmations"]
+                    .as_u64()
+                    .unwrap_or(DEFAULT_REQUIRED_CONFIRMATIONS as u64)
+            })
+            .into();
+
+        let nft_infos = get_nfts_for_activation(chain, &self.my_address, url).await?;
+
+        let global_nft = EthCoinImpl {
+            ticker,
+            coin_type: EthCoinType::Nft {
+                platform: self.ticker.clone(),
+            },
+            priv_key_policy: self.priv_key_policy.clone(),
+            my_address: self.my_address,
+            sign_message_prefix: self.sign_message_prefix.clone(),
+            swap_contract_address: self.swap_contract_address,
+            fallback_swap_contract: self.fallback_swap_contract,
+            contract_supports_watchers: self.contract_supports_watchers,
+            web3: self.web3.clone(),
+            web3_instances: self.web3_instances.clone(),
+            decimals: self.decimals,
+            gas_station_url: self.gas_station_url.clone(),
+            gas_station_decimals: self.gas_station_decimals,
+            gas_station_policy: self.gas_station_policy.clone(),
+            history_sync_state: Mutex::new(self.history_sync_state.lock().unwrap().clone()),
+            required_confirmations,
+            ctx: self.ctx.clone(),
+            chain_id: self.chain_id,
+            logs_block_range: self.logs_block_range,
+            nonce_lock: self.nonce_lock.clone(),
+            erc20_tokens_infos: Default::default(),
+            non_fungible_tokens_infos: Arc::new(AsyncMutex::new(nft_infos)),
+            abortable_system,
+        };
```

Comment:
you replace with this

```rust
let global_nft = EthCoinImpl {
            ticker,
            coin_type: EthCoinType::Nft {
                platform: self.ticker.clone(),
            },
            priv_key_policy: self.priv_key_policy.clone(),
            sign_message_prefix: self.sign_message_prefix.clone(),
            web3: self.web3.clone(),
            web3_instances: self.web3_instances.clone(),
            gas_station_url: self.gas_station_url.clone(),
            gas_station_policy: self.gas_station_policy.clone(),
            history_sync_state: Mutex::new(self.history_sync_state.lock().unwrap().clone()),
            required_confirmations,
            ctx: self.ctx.clone(),
            nonce_lock: self.nonce_lock.clone(),
            erc20_tokens_infos: Default::default(),
            non_fungible_tokens_infos: Arc::new(AsyncMutex::new(nft_infos)),
            abortable_system,
            ..**self
        };
        ```

Reply:
I also have a question here.
>Why did you get the value out of Mutex only to lock it in a Mutex again?

Reply:
> I also have a question here. `Why did you get the value out of Mutex only to lock it in a Mutex again? `

Sorry do you mean this line?
https://github.com/KomodoPlatform/komodo-defi-framework/blob/711651d17bf4bc56de3cb25bb7a551dd76883471/mm2src/coins/eth/v2_activation.rs#L321

We do it because we need to create a mutex with the inner value, we cant clone mutex itself.

We do the same when we enable erc20 token
https://github.com/KomodoPlatform/komodo-defi-framework/blob/711651d17bf4bc56de3cb25bb7a551dd76883471/mm2src/coins/eth/v2_activation.rs#L247

Reply:
I see..but we don't know the reason?

Reply:
> I see..but we don't know the reason?

I suppose the reason is bcz we build tokens from platform coin, so most of the fields should have the same state. The exceptions are `abortable_system`, `web3 instances`, `confirmations` etc - he fields that specific for token.

My note:  web3 use ticker name, but NFT_ETH, as an example, utilizes platform coin functionality, so I decided to clone it. confirmations have `AtomicU64` type, couldnt clone it.

Reply:
ok good. thanks

Reply:
> you can replace with this
```
            abortable_system,
            ..**self
        };
```

CI lint fails [comment](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2049#issuecomment-1900332499), lets not do it for now.

Reply:
ok ok